Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary/CdsMigratorPluginControl.cs

```diff
@@ -51,10 +51,12 @@ public override async void UpdateConnection(IOrganizationService newService, Con
                     var metaDataService = new MetadataService();
                     var exceptionService = new ExceptionService();
                     var viewHelpers = new ViewHelpers();
-                    this.importPage1.Tag = new ImportPagePresenter(this.importPage1, this, dataMigrationService, this, detail.ServiceClient, metaDataService, viewHelpers);
-                    this.exportPage1.Tag = new ExportPagePresenter(this.exportPage1, this, dataMigrationService, this, detail.ServiceClient, metaDataService, exceptionService, viewHelpers);
+                    this.importPage1.importPagePresenter = new ImportPagePresenter(this.importPage1, this, dataMigrationService, detail.ServiceClient, metaDataService, viewHelpers);
+                    this.exportPage1.exportPagePresenter = new ExportPagePresenter(this.exportPage1, this, dataMigrationService, detail.ServiceClient, metaDataService, exceptionService, viewHelpers);
```

Comment:
Also, the View shouldn't have access to the Presenter directly with MVP. The Presenter should be notified only by the events.

While we are using it directly yet, adding the public field in the view means it could be in the future. I realise this could be done with the Tag property but it would have to be casted, ringing the alarm bell. 

Therefore, I suggest ignoring my comment above and moving those fields to this class as private.

Reply:
So have `newImportPagePresenter(...)` and `newExportPagePresenter(...)` in the views stored as `Presenter` and `Presenter` variable is changed to private?

Reply:
So currently the presenters are "stored" on their respective views on the `Tag` property. This change has moved them to a typed and named field but still on the view. 

My understanding of the MVP pattern is that the presenter should not be accessible by the view, instead only via events. To achieve this, let's "store" the presenters on the parent component e.g. 
``` csharp
public partial class CdsMigratorPluginControl : PluginControlBase, IStatusBarMessenger
{
   private ImportPagePresenter ImportPagePresenter;
   private ExportPagePresenter ExportPagePresenter;

   ...
 
   this.ImportPagePresenter = new ImportPagePresenter(this.importPage1, this, dataMigrationService, detail.ServiceClient, metaDataService, viewHelpers);
   this.ExportPagePresenter = new ExportPagePresenter(this.exportPage1, this, dataMigrationService, detail.ServiceClient, metaDataService, exceptionService, viewHelpers);
}
```

Since it's private, the view won't have access. While the parent (`CdsMigratorPluginControl`) never uses it (only sets it) it avoids the garbage collector to dispose of the object.